docker/ubuntu.dockerfile

```diff
@@ -0,0 +1,150 @@
+        python3-venv \
+        wget \
+        build-essential \
+        git \
+        cmake \
+        gfortran \
+        libeigen3-dev \
+        libblas-dev \
+        liblapack-dev \
+        libhdf5-dev && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Create and activate virtual environment
+RUN python3 -m venv ${VENV_PATH} && \
+    ${VENV_PATH}/bin/pip install --upgrade pip && \
+    ${VENV_PATH}/bin/pip install \
+        numpy \
+        scipy \
+        cython \
+        pytest \
+        tables \
+        matplotlib \
+        jinja2 \
+        setuptools \
+        future \
+        progress
+
+ENV PATH=${VENV_PATH}/bin:$PATH \
+    LD_LIBRARY_PATH=${VENV_PATH}/lib:$LD_LIBRARY_PATH \
+    PYTHONPATH=${VENV_PATH}/lib/python3.12/site-packages:$PYTHONPATH
+
+# ------------------------------
+# Stage 2: MOAB Setup
+# ------------------------------
+FROM base AS moab
+
+ARG MOAB_VERSION=5.5.1
+ARG MAKE_CORES
+ARG INSTALL_PATH=/opt/moab
+
+RUN git clone --depth 1 --branch ${MOAB_VERSION} https://bitbucket.org/fathomteam/moab && \
+    cd moab && \
+    mkdir build && cd build && \
+    cmake .. \
+        -DENABLE_PYMOAB=ON \
+        -DCMAKE_INSTALL_PREFIX=${INSTALL_PATH} \
+        -DENABLE_HDF5=ON \
+        -DBUILD_SHARED_LIBS=ON \
+        -DENABLE_BLASLAPACK=OFF \
+        -DENABLE_FORTRAN=OFF && \
+    make -j ${MAKE_CORES} && \
+    make install && \
+    rm -rf $HOME/moab
+
+ENV PATH=${INSTALL_PATH}/bin:$PATH \
+    LD_LIBRARY_PATH=${INSTALL_PATH}/lib:$LD_LIBRARY_PATH \
+    PYTHONPATH=${INSTALL_PATH}/lib/python3.12/site-packages:$PYTHONPATH \
+    PYNE_MOAB_ARGS="--moab ${INSTALL_PATH}"
+
+# ------------------------------
+# Stage 3: DAGMC Setup
+# ------------------------------
+FROM moab AS dagmc
+
+ARG DAGMC_VERSION=v3.2.4
+ARG MAKE_CORES
+ARG INSTALL_PATH=/opt/dagmc
+
+RUN git clone --depth 1 --branch ${DAGMC_VERSION} https://github.com/svalinn/DAGMC.git && \
+    cd DAGMC && \
+    mkdir build && cd build && \
+    cmake .. \
+        -DMOAB_DIR=/opt/moab \
+        -DCMAKE_INSTALL_PREFIX=${INSTALL_PATH} \
+        -DBUILD_STATIC_LIBS=OFF \
+        -DBUILD_UWUW=OFF \
+        -DBUILD_TALLY=OFF \
+        -DBUILD_MAKE_WATERTIGHT=OFF \
+        -DBUILD_OVERLAP_CHECK=OFF \
+        -DBUILD_TESTS=OFF && \
+    make -j ${MAKE_CORES} && \
+    make install && \
+    rm -rf $HOME/DAGMC
+
+ENV PATH=${INSTALL_PATH}/bin:$PATH \
+    LD_LIBRARY_PATH=${INSTALL_PATH}/lib:$LD_LIBRARY_PATH \
+    PYNE_DAGMC_ARGS="--dagmc ${INSTALL_PATH}"
+
+# ------------------------------
+# Stage 4: OpenMC Setup
+# ------------------------------
+FROM dagmc AS openmc
+
+ARG OPENMC_VERSION=v0.15.2
+ARG MAKE_CORES
+ARG INSTALL_PATH=/opt/openmc
+
+RUN git clone --depth 1 --branch ${OPENMC_VERSION} https://github.com/openmc-dev/openmc.git && \
+    cd openmc && \
+    mkdir build && cd build && \
+    cmake .. \
+        -DCMAKE_INSTALL_PREFIX=${INSTALL_PATH} && \
+    make -j ${MAKE_CORES} && \
+    make install && \
+    cd .. && \
```

Comment:
I don't think we need to do a full install of OpenMC, just the python API....

Doesn't mean we can't do a full install, but for testing purposes we don't need it.